### DISCUSSION TITLE
AIO password field not declared as password

### DISCUSSION THREAD

**Discussion by MariusQuabeck** [nextcloud/all-in-one#2745] [Ideas] [answered]
<img width="644" alt="image" src="https://github.com/nextcloud/all-in-one/assets/17063564/8793bb38-4073-4d7a-b10c-3858b52ba205">

That field should be declared as a password field so that characters are hidden and the field can get picked up by password managers

**Comment by szaimen** [comment]
Hi, please read the discussion in https://github.com/nextcloud/all-in-one/commit/f518ea7feebd81d54a9697be9806adc63b37854b#commitcomment-106967941 and below

**Comment by szaimen** [comment, marked as answer]
@MariusQuabeck did you read it?

So the problem is basically that we want the password to be visible by default. However to make this work the input type must be text by default like it is currently. There is unfortunately no way to have both as far as I could find out: the input type being password by default and visible at the same time.

However if you find a way to do this I would gladly accept a PR! :)

**Reply by szaimen** [reply]
The solution was found with https://github.com/nextcloud/all-in-one/pull/2787

**Comment by szaimen** [comment]
I'll move this to ideas for now since it is the intended behaviour.

**Reply by rileychh** [reply]
@szaimen Thanks for explaining the intended behavior. I noticed the current implementation in #2787 has the fields start as password type but automatically switch to text type as soon as you type anything, which feels a bit unexpected from a UX perspective.

Would it make more sense to follow the common pattern of having a show/hide toggle button next to the password field? That way:
- Password stays masked by default (better security)
- Users can click to show it when needed (better usability)
- Behavior matches what users expect from other websites

Something simple like:
```html
<input type="password" id="master-password">
<button type="button" onclick="togglePasswordVisibility('master-password')">
  Show/Hide
</button>
```

With JavaScript:
```javascript
function togglePasswordVisibility(id) {
  const field = document.getElementById(id);
  field.type = field.type === "password" ? "text" : "password";
}
```

This puts the user in control rather than having the password automatically reveal itself. What do you think?
